sos/deviceauth.py

```diff
@@ -0,0 +1,291 @@
+# Copyright (C) 2023 Red Hat, Inc., Jose Castillo <jcastillo@redhat.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+import logging
+import requests
+import time
+from datetime import datetime, timedelta
+import os
+from sos.keyring import Key, KeyNotFoundError
+import shutil
+
+DEVICE_AUTH_CLIENT_ID = "sos-tools"
+GRANT_TYPE_DEVICE_CODE = "urn:ietf:params:oauth:grant-type:device_code"
+DATETIME_FORMAT = "%Y-%m-%dT%H:%M:%SZ"
+
+logger = logging.getLogger("sos")
+
+
+def try_read_refresh_token():
+    """Try to read locally stored refresh token
+
+    Returns:
+        str: Returns ODIC refresh token if found otherwise None
+    """
+    try:
+        RHELKey = Key()
```

Comment:
I dont like much explicit distribution name class (*RHEL*Key) being used in generic code (here and any further). I understand this code will be executed only on RHEL distro, but some implicit structure of the classes or files is broken.

We can add a generic class layer that will be implemented only in RHELKey. Or maybe we can put RHELKey into `sos.policie.distros`? Where also `self.token_endpoint` and similar `redhat.com` strings would fit the best..?

Reply:
I think the problem is that the name should be more generic, since this is not a RHEL specific key, but rather a kernel keyring key. Would changing the name help?